Classes/Domain/Repository/BackendUserRepository.php

```diff
@@ -2,69 +2,34 @@
- * of the License, or any later version.
- *
- * For the full copyright and license information, please read the
- * LICENSE.txt file that was distributed with this source code.
- *
- * The TYPO3 project - inspiring people to share!
- */
-use TYPO3\CMS\Extbase\Persistence\QueryResultInterface;
-use TYPO3\CMS\Backend\Utility\BackendUtility;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Persistence\Generic\Typo3QuerySettings;
+use TYPO3\CMS\Extbase\Persistence\Repository;
 
-/**
- * Repository for \TYPO3\CMS\Extbase\Domain\Model\BackendUser.
- *
- * @api
- */
-class BackendUserRepository extends \TYPO3\CMS\Extbase\Domain\Repository\BackendUserRepository
+class BackendUserRepository extends Repository
 {
-
-    /**
-     * get uncached usergroups of be_user.usergroup for newsinvitations
-     *
-     * @param array $usergroups
-     *
-     * @return array|QueryResultInterface
-     */
-    public function findByUsergroups($usergroups)
+    public function initializeObject()
     {
-        $usergroupConstraints = [];
-        foreach ($usergroups as $usergroup) {
-            $usergroupConstraints[] = 'AND FIND_IN_SET(' . $usergroup . ', usergroup) ';
-        }
-        $statement = 'SELECT * FROM be_users WHERE 1=1 ' . implode($usergroupConstraints) . BackendUtility::BEenableFields('be_users') . BackendUtility::deleteClause('be_users');
-        $query = $this->createQuery();
-        $query->statement($statement);
-
-        return $query->execute(true);
+        $querySettings = GeneralUtility::makeInstance(Typo3QuerySettings::class);
+        $querySettings->setRespectStoragePage(false);
+        $this->setDefaultQuerySettings($querySettings);
     }
 
-    /**
-     * get cached usergroups of be_user.usergroup_cached_list
-     *
-     * attention: setCachedList() is only up to date if a user has loged in after groups have been added/removed!
-     * see: web/typo3/sysext/core/Classes/Authentication/BackendUserAuthentication::setCachedList()
-     *
-     * @param array $usergroups
-     *
-     * @return array|QueryResultInterface
-     */
-    public function findByUsergroupsCached($usergroups)
+    public function findByUsergroups(array $usergroups): array
     {
-        $usergroupConstraints = [];
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getConnectionForTable('be_users')->createQueryBuilder();
+
+        $constraints = [];
         foreach ($usergroups as $usergroup) {
-            $usergroupConstraints[] = 'AND FIND_IN_SET(' . $usergroup . ', usergroup_cached_list) ';
+            $constraints[] = $queryBuilder->expr()->inSet('bu.usergroup', $queryBuilder->createNamedParameter((int)$usergroup, \PDO::PARAM_INT));
         }
-        $statement = 'SELECT * FROM be_users WHERE 1=1 ' . implode($usergroupConstraints) . BackendUtility::BEenableFields('be_users') . BackendUtility::deleteClause('be_users');
```

Comment:
I assume enableFields and deleteClause are respected automatically in the queryBuilder?

Reply:
Yes, that's the case. @lorenzulrich